datanode/sqlstore/sqlstore.go

```diff
@@ -38,21 +38,57 @@ var EmbedMigrations embed.FS
 
 const SQLMigrationsDir = "migrations"
 
-func MigrateSchema(log *logging.Logger, config ConnectionConfig) error {
+func MigrateToLatestSchema(log *logging.Logger, config Config) error {
+	migrate := func(db *sql.DB) error {
+		if err := goose.Up(db, SQLMigrationsDir); err != nil {
+			return fmt.Errorf("error migrating sql schema: %w", err)
+		}
+
+		return nil
+	}
+
+	err := runMigration(log, config, migrate)
+	if err != nil {
+		return fmt.Errorf("failed to run migration to latest schema:%w", err)
+	}
+
+	return nil
+}
+
+func MigrateToSchemaVersion(log *logging.Logger, config Config, version int64) error {
+	migrate := func(db *sql.DB) error {
```

Comment:
I don't think we need to create these inline functions everywhere, can they not just look like this:
```
func MigrateToSchemaVersion() {
    db := getDbThing()
    defer db.Close()

    if err := goose.UpTo(db, SQLMigrationsDir, version); err != nil {
        return fmt.Errorf("error migrating sql schema: %w", err)
    }
}
```
and

```
func MigrateToLatestSchema() {
    db := getDbThing()
    defer db.Close()

    if err := goose.Up(db, SQLMigrationsDir); err != nil {
        return fmt.Errorf("error migrating sql schema: %w", err)
    }
}
```
where `getDbThing()` is the start of `runMigration`

Reply:
done this was to remove a bit of duplicaton of db creation and goose setup, but arguagbly the complexity not worth the biscuit, just duplicated it instead, encapuslation of db creation saves one line so not really worth it.